linuxkm/lkcapi_sha_glue.c

```diff
@@ -944,24 +971,34 @@ static int wc_linuxkm_drbg_generate(struct crypto_rng *tfm,
 {
     struct wc_linuxkm_drbg_ctx *ctx = (struct wc_linuxkm_drbg_ctx *)crypto_rng_ctx(tfm);
     int ret;
-
-    wc_LockMutex(&ctx->lock);
+    int my_cpu =
+        raw_smp_processor_id(); /* Note, core is not locked, so the actual core
+                                 * ID may change while executing, hence the
+                                 * mutex.
+                                 * The mutex is also needed to coordinate with
+                                 * wc_linuxkm_drbg_seed(), which seeds all
+                                 * instances.
+                                 */
+    wolfSSL_Mutex *lock = &ctx->rngs[my_cpu].lock;
+    WC_RNG *rng = &ctx->rngs[my_cpu].rng;
+
+    wc_LockMutex(lock);
```

Comment:
Check on return values of wc_LockMutex

Reply:
Fair enough, re smallstack path -- I'll special-case it for `WOLFSSL_LINUXKM`.  We indeed don't want to have heap ops on the kernel for SHA/HMAC generator ops, and can afford the stack space, but it's a super-basic op and the extra 32 bytes might matter on some embedded targets.

Linux kernel mutex lock is a `void`, but our wrapper isn't and there's no harm adding an error check.  Will do.

Reply:
@JacobBarthelmeh, both items are fixed in code just pushed.